code/script.js

```diff
@@ -29,23 +40,247 @@ const showMessage = (message, sender) => {
+        console.log("area Choice is:", areaChoice);
+        setTimeout(botQuestion5, 1000);
+        break;
+      case "eastAfrica":
+        showMessage(`East Africa and the Serengeti`, 'user');
+        console.log("area Choice is:", areaChoice);
+        setTimeout(botQuestion5, 1000);
+        break;
+      case "southAfrica":
+        showMessage(`Southern Africa, Namib Desert, and The Cape of Good Hope`, 'user');
+        console.log("area Choice is:", areaChoice);
+        setTimeout(botQuestion5, 1000);
+        break;
+      case "britishIsles":
+        showMessage(`The British Isles, and Ireland`, 'user');
+        console.log("area Choice is:", areaChoice);
+        setTimeout(botQuestion5, 1000);
+        break;
+      case "scandinavia":
+        showMessage(`Scandinavia, The Baltic and North Sea`, 'user');
+        console.log("area Choice is:", areaChoice);
+        setTimeout(botQuestion5, 1000);
+        break;
+      case "theAlps":
+        showMessage(`Central Uplands, The Alps and the Mediterranean`, 'user');
+        console.log("area Choice is:", areaChoice);
+        setTimeout(botQuestion5, 1000);
+        break;
+    }
+  }, { once: true });  //function only used once
+}
+
+
+//---------------- Bot Question 5 - Date to travel -------------------------//
+
+const botQuestion5 = () => {
+  console.log("You have entered Question 5 function"); //reference for me to help me track the flow
+  showMessage(`We have some amazing locations in this area`, 'bot');
+  showMessage(`Please use the date picker on the right to choose the date you wish to travel`, 'bot');
+
+  //create date input field to choose date
+  const chooseDate = document.getElementById("input-wrapper")
+  chooseDate.innerHTML = `
+    <input type="date" id="trip-start" name="trip-start" value="2023-09-03" min="2023-09-03" max="2024-12-31" />
+    `
+  //document.getElementById("name-input").readOnly = true; My attempt at getting the input field for the date working. At this stage the date field only works when the user users the date picker and NOT when they enter a date into the input field//
+
+  //event listener for when user input's date
+  chooseDate.addEventListener("change", function (whichButton) {
+    let usersDate = document.getElementById("trip-start").value;
+    console.log(usersDate)
+    showMessage(`${usersDate}`, 'user');
+    setTimeout(botQuestion6, 1000);
+  })
+}
+
+
+//---------------- Bot Question 6 - Confirmation to proceed -------------------------//
+
+const botQuestion6 = () => {
+  console.log("You have entered Question 6 function"); //reference for me to help me track the flow
+  showMessage(`Our team now have enough information to prepare some options for you`, 'bot');
+  showMessage(`We will leave you with a reference number that will help you access more details for your journey`, 'bot');
+  setTimeout(showMessage(`Would you like as to procceed?`, 'bot'), 1000); //this setTimeout doesn't seem to work?
```

Comment:
This is the only place where "showMessage" is within the setTimeout. Otherwise, we link to the next function which contains the message. Maybe that is why?